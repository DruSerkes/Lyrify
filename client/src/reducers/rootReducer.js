import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
