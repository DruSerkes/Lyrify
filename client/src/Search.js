import React from 'react';
import { Box, Typography } from '@material-ui/core';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { getSong } from './reducers/actions';
import SearchForm from './SearchForm';
import Lyrics from './Lyrics';

const Search = () => {
	const dispatch = useDispatch();
	const songData = useSelector((state) => state.song, shallowEqual);
	const doSearch = (data) => dispatch(getSong(data));

	return (
		<Box padding={1} margin={1}>
			<Box padding={1} margin={1} className="Search-Header">
				<Typography variant="h1">Search Lyrics</Typography>
				<SearchForm doSearch={doSearch} />
			</Box>
			<Box component="section" className="Lyrics">
				{songData ? <Lyrics songData={songData} /> : null}
			</Box>
		</Box>
	);
};

export default Search;
