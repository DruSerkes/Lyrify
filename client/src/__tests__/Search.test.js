import React from 'react';
import { render } from '@testing-library/react';
import Search from '../Search';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import 'mutationobserver-shim';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
const songData = {
	id     : 1,
	song   : 'test song',
	artist : 'the testers'
};

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Search songData={songData} />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Search songData={songData} />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
