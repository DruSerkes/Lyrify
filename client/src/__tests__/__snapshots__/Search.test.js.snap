// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-9"
  >
    <div
      class="MuiBox-root MuiBox-root-10 Search-Header"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Search Lyrics
      </h1>
      <form
        class="SearchForm"
      >
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="song"
              id="song-label"
            >
              Song Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="song"
                name="song"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-13 PrivateNotchedOutline-legendNotched-14"
                >
                  <span>
                    Song Name *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="artist"
              id="artist-label"
            >
              Artist Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="artist"
                name="artist"
                required=""
                type=""
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-13"
                >
                  <span>
                    Artist Name *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
    <section
      class="MuiBox-root MuiBox-root-15 Lyrics"
    >
      <div
        class="MuiPaper-root Lyrics MuiPaper-elevation5 MuiPaper-rounded"
      >
        <section
          class="Lyrics-Content"
        >
          <div
            class="Lyrics-Header"
          >
            <div
              class="Lyrics-Header-Top"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                "test song"
              </h4>
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                By: the testers
              </h5>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle"
          />
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Cannot find lyrics for this song
            </p>
          </div>
        </section>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
